Orange/widgets/unsupervised/owsom.py

```diff
@@ -345,6 +344,12 @@ def set_warnings():
             else:
                 prepare_data()
 
+        invalidated = not (cont_x is not None and self.cont_x is not None
+                           and array_equal(cont_x, self.cont_x))
```

Comment:
I would find `cont_x is None or self.cont_x is None or not array_equal(cont_x, self.cont_x)` more readable.

I understand what you meant ("it is not invalidated if both are non-None and equal"), but I find this difficult to process.

There's also an edge case: if both are `None`, `invalidated` should be `False`. This could be covered by `(cont_x is None) != (self.cont_x is None) or (cont_x is not None and array_equal(cont_x, self.cont_x))`, but there's no need for because nothing bad happens in that case.

Reply:
Done.